
#### Regresión Lineal Múltiple: Influencia del Estilo de Vida sobre la Glucosa en Ayunas

La aparición y progresión de la diabetes mellitus están fuertemente influenciadas por factores de estilo de vida, tales como la actividad física, el consumo de alcohol y el estado nutricional. 

El objetivo de este experimento es analizar cuantitativamente la relación entre los factores de estilo de vida y los niveles de glucosa en ayunas (`glucose_fasting`), utilizando un modelo de regresión lineal múltiple ya que la glucosa en ayunas es un indicador clínico ampliamente utilizado para la detección temprana de alteraciones metabólicas y estados prediabéticos. Las variables fueron seleccionadas debido a su relación documentada con el riesgo metabólico y la aparición de diabetes.

#### Metodología

Se ajustó un modelo de **regresión lineal múltiple** mediante el método de **mínimos cuadrados ordinarios (OLS)**. Previamente, se incorporó un término constante para modelar el intercepto. El análisis fue realizado utilizando la librería `statsmodels` de Python.



In [49]:
import statsmodels.api as sm
import pandas as pd

# Cargar el dataset
df = pd.read_csv('Diabetes_and_LifeStyle_Dataset .csv')


# Variables de estilo de vida seleccionadas
vars = [
    "Age",
    "alcohol_consumption_per_week",
    "physical_activity_minutes_per_week",
    "diet_score",
    "screen_time_hours_per_day",
]

# Definir variables dependiente e independientes
X = df[vars]
Y = df["glucose_fasting"]

# Agregar constante
X = sm.add_constant(X)

# Ajustar modelo
model_lifestyle = sm.OLS(Y, X).fit()

# Mostrar resultados
print(model_lifestyle.summary())

                            OLS Regression Results                            
Dep. Variable:        glucose_fasting   R-squared:                       0.086
Model:                            OLS   Adj. R-squared:                  0.086
Method:                 Least Squares   F-statistic:                     1838.
Date:                Thu, 08 Jan 2026   Prob (F-statistic):               0.00
Time:                        17:54:17   Log-Likelihood:            -3.8755e+05
No. Observations:               97297   AIC:                         7.751e+05
Df Residuals:                   97291   BIC:                         7.752e+05
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
                                         coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------------------
cons

El modelo de regresión lineal ajustado para evaluar la relación entre variables de estilo de vida y la glucosa en ayunas (`glucose_fasting`) mostró un R² ajustado de 0.086, indicando que aproximadamente el 8.6% de la variabilidad en la glucosa se explica por las variables incluidas en el modelo. Entre las variables consideradas, la edad presentó un efecto positivo significativo, indicando que la glucosa tiende a aumentar con la edad. El consumo de alcohol mostró un efecto positivo moderado, mientras que la actividad física y la puntuación de dieta presentaron efectos negativos significativos, lo que sugiere que mayores niveles de actividad y mejores hábitos alimenticios se asocian con glucosa más baja. El tiempo frente a la pantalla tuvo efectos positivos, indicando que un mayor sedentarismo se relacionan con niveles más altos de glucosa. 

Estos resultados reflejan que, aunque el estilo de vida influye en los niveles de glucosa, su capacidad explicativa es limitada, y factores adicionales como antecedentes clínicos, niveles de insulina o HbA1c serían necesarios para construir un modelo predictivo más robusto. Desde un punto de vista práctico, el análisis sugiere que las intervenciones centradas en aumentar la actividad física, mejorar la calidad de la dieta y reducir el sedentarismo podrían contribuir a mantener niveles de glucosa más saludables y, potencialmente, a disminuir el riesgo de desarrollar diabetes. El signo y la significancia de los coeficientes observados son consistentes con la evidencia clínica existente y reflejan patrones esperados de riesgo metabólico.

Además, el *condition number* reducido a 873 indica que no existen problemas graves de multicolinealidad entre estas variables, asegurando la confiabilidad de los coeficientes estimados.
